File: src/main/java/com/informationsystemshoestore/shoestoredataservice/routing/StatisticRouter.java
package com.informationsystemshoestore.shoestoredataservice.routing;

import com.informationsystemshoestore.shoestoredataservice.handler.StatisticHandler;
import com.informationsystemshoestore.shoestoredataservice.routing.path.StatisticPath;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.RouterFunctions;
import org.springframework.web.reactive.function.server.ServerResponse;

@Configuration
public class StatisticRouter {

    @Bean
    public RouterFunction<ServerResponse> statisticRoute(StatisticHandler handler) {

        return RouterFunctions
                .route(StatisticPath.EXPORT_STATISTIC, handler::exportCountStatistic);
    }
}

File: src/main/kotlin/com/informationsystemshoestore/shoestorestatisticsservice/router/StatisticRouter.kt
package com.informationsystemshoestore.shoestorestatisticsservice.router

import com.informationsystemshoestore.shoestorestatisticsservice.handler.StatisticHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType.APPLICATION_JSON
import org.springframework.web.reactive.function.server.coRouter

@Configuration
class StatisticRouter(private val statisticHandler: StatisticHandler) {

    @Bean
    fun apiRouter() = coRouter {
        "/api/v1/statistics".nest {
            accept(APPLICATION_JSON).nest {
                GET("", statisticHandler::getStatistic)
            }
        }
    }
}